### DISCUSSION TITLE
During the step "Replacing all properties environment-related" is it possible to only replace SOME Global Paramaters?

### DISCUSSION THREAD

**Discussion by ghost** [Azure-Player/azure.datafactory.tools#303] [Q&A] [answered]
I have Terraform that creates our ADF in Dev/Qa/Uat/Prod. It also creates some specific global parameters for each of those environments. 

Although, my ADF repo also needs to create Global Parameters with specific values for each environment, BUT I want it to leave other values alone. (The ones I created via Terraform.)

Is this possible?

Perhaps, I can just take them out of the "factory" json file??

Or is there some way to specify in the parameter file to keep the existing value. Like:

factory, DevFactory, "$.properties.globalParameters.'Env-Code'.value", $ExistingValue

**Comment by NowinskiK** [comment, marked as answer]
No, it's not possible.
Global parameters can be deploy ALL or nothing (excluded). This is due to how API is built.
Also, there is no option to keep the existing values - the deployment process doesn't merge objects.
I would suggest to keep complete list of global parameters in the repo. It shouldn't be part of Infra deployment.

**Comment by sandeeppatidar76** [comment]
@shaynej-jackson , I have done some workaround to deploy selected Global parameters, this may not be the right way to do it. but you can achieve your purpose.
So for this you can deploy global parameter by updating you config CSV file, Remove all those parameter which you want to deploy,like below
factory,devadf,-$.properties.globalParameters.fileshareuser,
